java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,691 @@
+        build(e.expression());
+        break;
+      case PARENTHESIZED_EXPRESSION:
+        build(((ParenthesizedTree) tree).expression());
+        break;
+      case ARRAY_ACCESS_EXPRESSION: {
+        ArrayAccessExpressionTree aaet = (ArrayAccessExpressionTree) tree;
+        currentBlock.elements.add(aaet);
+        build(aaet.expression());
+        build(aaet.dimension());
+        break;
+      }
+      case ARRAY_DIMENSION: {
+        ArrayDimensionTree arrayDimensionTree = (ArrayDimensionTree) tree;
+        if(arrayDimensionTree.expression() != null) {
+          build(arrayDimensionTree.expression());
+        }
+        break;
+      }
+      case NEW_CLASS: {
+        NewClassTree newClassTree = (NewClassTree) tree;
+        currentBlock.elements.add(newClassTree);
+        if(newClassTree.classBody() != null) {
+          build(newClassTree.classBody());
+        }
+        build(newClassTree.identifier());
+        if(newClassTree.enclosingExpression() != null) {
+          build(newClassTree.enclosingExpression());
+        }
+        build(Lists.reverse(newClassTree.arguments()));
+      }
+      break;
+      case METHOD_REFERENCE:
+        MethodReferenceTree methodReferenceTree = (MethodReferenceTree) tree;
+        build(methodReferenceTree.method());
+        if(methodReferenceTree.typeArguments() != null) {
+          build(Lists.reverse(methodReferenceTree.typeArguments()));
+        }
+        build(methodReferenceTree.expression());
+        break;
+      case IDENTIFIER:
+      case INT_LITERAL:
+      case LONG_LITERAL:
+      case DOUBLE_LITERAL:
+      case CHAR_LITERAL:
+      case FLOAT_LITERAL:
+      case STRING_LITERAL:
+      case BOOLEAN_LITERAL:
+      case NULL_LITERAL:
+      case PRIMITIVE_TYPE:
+        currentBlock.elements.add(tree);
+        break;
+      case UNBOUNDED_WILDCARD:
+      case EXTENDS_WILDCARD:
+      case SUPER_WILDCARD:
+        //Ignore bounds in CFG
+        currentBlock.elements.add(tree);
+        break;
+      case TYPE_CAST:
+        currentBlock.elements.add(tree);
+        TypeCastTree typeCastTree = (TypeCastTree) tree;
+        build(typeCastTree.type());
+        build(typeCastTree.expression());
+        break;
+      case PARAMETERIZED_TYPE:
+        ParameterizedTypeTree parameterizedTypeTree = (ParameterizedTypeTree) tree;
+        currentBlock.elements.add(parameterizedTypeTree);
+        build(parameterizedTypeTree.type());
+        build(Lists.reverse(parameterizedTypeTree.typeArguments()));
+        break;
+      case INSTANCE_OF:
+        InstanceOfTree instanceOfTree = (InstanceOfTree) tree;
+        currentBlock.elements.add(instanceOfTree);
+        build(instanceOfTree.type());
+        build(instanceOfTree.expression());
+        break;
+      case EMPTY_STATEMENT:
+        break;
+      case COMPILATION_UNIT:
+        break;
+      case CLASS:
+      case INTERFACE:
+      case ENUM:
+      case ANNOTATION_TYPE:
+      case LAMBDA_EXPRESSION:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This case's code block is the same as the block for the case on line 531. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1871)